catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/ToIcebergSortOrder.java

```diff
@@ -58,9 +59,10 @@ public static org.apache.iceberg.SortOrder toSortOrder(Schema schema, SortOrder[
 
             Expression firstArg = sortFunc.arguments()[0];
             Preconditions.checkArgument(
-                firstArg instanceof Literal && ((Literal<?>) firstArg).value() instanceof Integer,
+                firstArg instanceof Literal
+                    && ((Literal<?>) firstArg).dataType() instanceof Types.IntegerType,
                 "Bucket sort's first argument must be a integer literal");
-            int numBuckets = (Integer) ((Literal<?>) firstArg).value();
+            int numBuckets = Integer.parseInt(String.valueOf(((Literal<?>) firstArg).value()));
```

Comment:
![image](https://github.com/datastrato/gravitino/assets/94670132/8dfa3cad-c7dd-4d3e-b928-c8f8e12bfc3c)

Reply:
Can you provide the entry point for calling the function in the picture? I think the value here should not be a string.

Reply:
Oh, I see. Literal will keep the string value to postpone the literal value calculation to the real catalog engine. 

So it is fine for this part changes

Reply:
> Oh, I see. Literal will keep the string value to postpone the literal value calculation to the real catalog engine.
> 
> So it is fine for this part changes

ok.